crates/yas_periphery/src/yas_nft_position_manager.cairo

```diff
@@ -1,24 +1,635 @@
+    ) -> bool;
 }
 
 #[starknet::contract]
 mod YASNFTPositionManager {
-    use super::IYASNFTPositionManager;
+    use super::{
+        IYASNFTPositionManager, AddLiquidityParams, IncreaseLiquidityParams,
+        DecreaseLiquidityParams, MintParams, Position, PoolKey
+    };
+    use starknet::{
+        ContractAddress, get_contract_address, contract_address_const, get_caller_address
+    };
+    use openzeppelin::token::erc721::ERC721;
 
-    use yas_core::numbers::signed_integer::{i32::i32, integer_trait::IntegerTrait};
+    use yas_core::contracts::yas_pool::{IYASPoolDispatcher, IYASPoolDispatcherTrait};
+    use yas_core::contracts::yas_factory::{IYASFactoryDispatcher, IYASFactoryDispatcherTrait};
+    use yas_core::interfaces::interface_ERC20::{IERC20Dispatcher, IERC20DispatcherTrait};
+    use yas_core::libraries::position::{PositionKey, Info};
+    use yas_core::libraries::tick_math::TickMath::{
+        get_tick_at_sqrt_ratio, get_sqrt_ratio_at_tick, MIN_TICK, MAX_TICK
+    };
+    use yas_core::numbers::fixed_point::implementations::impl_64x96::{
+        FixedType, FixedTrait, FP64x96PartialOrd, FP64x96PartialEq, FP64x96Impl, FP64x96Zeroable,
+        FP64x96Sub, ONE
+    };
+    use yas_core::numbers::signed_integer::{i32::i32};
+    use yas_core::utils::math_utils::Constants::Q128;
+    use yas_core::utils::math_utils::FullMath;
+    use yas_core::utils::utils::ContractAddressPartialOrd;
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        IncreaseLiquidity: IncreaseLiquidity,
+        DecreaseLiquidity: DecreaseLiquidity,
+        Collect: Collect,
+        Approval: Approval,
+        MintCallback: MintCallback
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct IncreaseLiquidity {
+        token_id: u256,
+        liquidity: u128,
+        amount_0: u256,
+        amount_1: u256
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct DecreaseLiquidity {
+        token_id: u256,
+        liquidity: u128,
+        amount_0: u256,
+        amount_1: u256
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct Collect {
+        token_id: u256,
+        recipient: ContractAddress,
+        amount_0_collect: u256,
+        amount_1_collect: u256
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct Approval {
+        owner: ContractAddress,
+        to: ContractAddress,
+        token_id: u256,
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct MintCallback {
+        amount_0_owed: u256,
+        amount_1_owed: u256
+    }
 
     #[storage]
-    struct Storage {}
+    struct Storage {
+        factory: ContractAddress,
+        pool_ids: LegacyMap<ContractAddress, u128>,
+        pool_id_to_pool_key: LegacyMap<u128, PoolKey>,
+        positions: LegacyMap<u256, Position>,
+        next_id: u256,
+        next_pool_id: u128
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, factory: ContractAddress) {
+        // TODO: ERC721Permit.sol
```

Comment:
Aren't we going to avoid this implementation?

Reply:
You're right, this comment is not necessary since we won't be using that implementation anymore

Reply:
✔️